File: 1310 XOR Queries of a Subarray/1310.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        for i in range(1, len(arr)):
            arr[i] ^= arr[i-1]
        
        ans = []

        for i, j in queries:
            if i == 0:
                ans.append(arr[j])
            else:
                ans.append(arr[i-1] ^ arr[j])

        return ans

File: 1310 XOR Queries of a Subarray/1310.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        for (int i = 1; i < arr.size(); i++){
            arr[i] ^= arr[i-1];
        }

        vector<int> ans;

        for (vector<int>& q: queries){
            int i = q[0];
            int j = q[1];
            if (i == 0){
                ans.push_back(arr[j]);
            } else {
                ans.push_back(arr[i-1] ^ arr[j]);
            }
        }

        return ans;
    }
};
